### ISSUE TITLE
Reduce image hosting costs for server owners.

### ISSUE THREAD

**Issue by TheOneImprove** [LemmyNet/lemmy#5027]
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?

### Is your proposal related to a problem?

Provide solutions to reducing image hosting for the server.

### Describe the solution you'd like.

Provide hash deduplication function for images with control for the server owner in the database, by comuting the image hash on the user device while uploading it, while linking the images to the posts.

in simple terms, instead of hosting the same image twice, it get hosted once.

### Describe alternatives you've considered.

- Integrate imgur to upload the image to it while saving the delete url for the image for the user account.

**Comment by Nutomic** [comment]
pict-rs already deduplicates images if they are identical.
